scrapy/core/scraper.py

```diff
@@ -147,6 +147,14 @@ def _scrape2(self, request_result, request, spider):
 
     def call_spider(self, result, request, spider):
         result.request = request
+        # StopDownload exceptions: make the partial response an attribute of the failure
+        if (
+            isinstance(result, Failure)
+            and isinstance(result.value, StopDownload)
+            and hasattr(result.value, "response")
+        ):
+            result.response = result.value.response
+            delattr(result.value, "response")
```

Comment:
Why not `del result.value.response`?

Reply:
No reason in particular, I just didn't think of it. Do you think `del` is more readable than `dellattr`? I don't really have a strong opinion on this.

Reply:
I think the main difference is that delattr allows arbitrary string variables, while del needs a Python expression. So for me delattr here feels like using `setattr(foo, "bar", 5)` instead of `foo.bar = 5`. 

On the other hand, with `delattr(result.value, "response")` is is more clear that we're deleting the response attribute, while with `del result.value.response` it is not immediately clear what's being deleted.

So that's fine to keep it as-is as well :)